### PR TITLE
Assorted Changes 

### PR DESCRIPTION
**Pull request by sleitner**
Some clean up, new tag functionality in subject_matter field, and empty value radio buttons by default.

### CODE REVIEW

File: ckanext/cfpb_extrafields/plugin.py [hunk 1]
```diff
@@ -148,6 +146,8 @@ def _modify_package_schema(self, schema):
                         tk.get_converter('convert_to_extras')],
 
         })
+        # this update is for special fields that become free tags AND normal fields.
```
Comment (Ooblioob) [inline ckanext/cfpb_extrafields/plugin.py:149]: Good stuff, thanks for adding the comment! :+1:

File: ckanext/cfpb_extrafields/validators.py [hunk 2]
```diff
@@ -6,11 +6,6 @@ def Invalid(message):
     import ckan.plugins.toolkit as tk
     raise tk.Invalid(message)
 
-def required_field(value):
```
Comment (Ooblioob) [inline ckanext/cfpb_extrafields/validators.py:9]: oh good, I didn't like this method anyway.  Does that mean we found a built-in substitute for it?

**Comment by Ooblioob** [comment]
Very good, this all looks great to me.  No objections.  Nice work @sleitner!

File: ckanext/cfpb_extrafields/validators.py (hunk 2 above)
Comment (sleitner) [inline ckanext/cfpb_extrafields/validators.py:9]: thanks for the review, Dan! Yup, it was a little hidden away but there was, indeed, a 'not_empty'

**Pull request merged**